extensions/interactions/FractionInput/directives/fraction-input-validation.service.spec.ts

```diff
@@ -244,22 +249,37 @@ describe('FractionInputValidationService', () => {
   });
 
   it('should not catch equals followed by equivalent as redundant', function() {
-    answerGroups[0].rules = [equalsOneRule, equivalentToOneRule];
-    var warnings = validatorService.getAllWarnings(
-      currentState, customizationArgs, answerGroups,
+    const answerGroup1 = agof.createNew(
+      {}, goodDefaultOutcome, false, null);
+    answerGroup1.updateRuleInputs([equalsOneRule]);
+
+    const answerGroup2 = agof.createNew(
+      {}, goodDefaultOutcome, false, null);
+    answerGroup1.updateRuleInputs([equivalentToOneRule]);
```

Comment:
The concept of "followed by" doesn't apply to rules within an answer group anymore, only between answer groups. The old test had both rules in one answer group, and didn't mark the equalsOneRule as redundant because it was before the equivalentToOneRule. However, now that rules don't have an order, the old test was failing because it was marking equalsOneRule as redundant (which it technically is, since rules are OR'ed together within an AnswerGroup).

I decided to keep this behavior, and to support this, I made changes to the validation service so that for rules within an answer group, it will check both ways, and for rules that aren't, it would only check one direction. The test above needed to change, however, for the case that the rules are in separate answer groups. Hope that makes sense.